Week3/akif-week3.js

```diff
@@ -0,0 +1,158 @@
+//   STRINGS! --------------
+let myString = 'hello,this,is,a,difficult,to,read,sentence';
+console.log(myString);
+console.log(myString.length);
+myString = myString.replace(/[,]/g, ' ');
+console.log(myString);
+
+//   ARRAYS! -------------------------------------------------
+let favoriteAnimals = ['blowfish', 'capricorn', 'giraffe'];
+favoriteAnimals.push('turtle');
+console.log(favoriteAnimals);
+favoriteAnimals.splice(1, 0, 'meerkat');
+console.log("i added a new value 'meerkat' to the array");
+console.log(favoriteAnimals);
+console.log('The array has a length of: ' + favoriteAnimals.length);
+favoriteAnimals.splice(3, 1);
+console.log(favoriteAnimals);
+console.log('The item you are looking for is at index: ' + favoriteAnimals.indexOf('meerkat'));
+favoriteAnimals.splice(favoriteAnimals.indexOf('meerkat'), 1);
```

Comment:
This is correct but you will learn a little later that using splice actually changes the favoriteAnimals array itself. This is called mutating the data structure and often it is suggested that we don't mutate the data structure i.e try to push the meerkat in between the two elements but return a new array itself thus having two arrays at the end, original and the modified array.

Maybe when you are introduced to this concept, you can revisit it and think of a way by which you can still add meerkat but don't modify the existing array. 🙂

Reply:
Hi Yash. You mean something like  this ?
let a = favoriteAnimals.slice(1, favoriteAnimals.length);
let b = favoriteAnimals.slice(0, 1);
b.push('meerkat');
b.push(...a);

Reply:
Yes. Something similar 🙂